torch/lib/c10d/ProcessGroupMPI.cpp

```diff
@@ -91,6 +92,73 @@ void checkSameSizeAndType(
   }
 }
 
+void checkSplitSizes(
+    const std::vector<int64_t>& split_sizes,
+    const at::Tensor& tensor,
+    int group_size) {
+  if (split_sizes.size() == 0) {
+    if (tensor.size(0) % group_size != 0) {
+      throw std::runtime_error(
+          "Tensor's dim 0 does not divide equally across group size");
+    }
+  } else {
+    if (split_sizes.size() != group_size) {
+      throw std::runtime_error("Number of tensor splits not equal to group size");
+    }
+    int sum = std::accumulate(split_sizes.begin(), split_sizes.end(), 0);
+    if (sum != tensor.size(0)) {
+      throw std::runtime_error("Split sizes doesn't match total dim 0 size");
+    }
+  }
+}
+
+int64_t computeLengthsAndOffsets(
+    const std::vector<int64_t>& split_sizes,
+    const at::Tensor& tensor,
+    std::vector<int>& lengths,
+    std::vector<int>& offsets,
```

Comment:
nit: since these are output params, do you mind making them last and having 
std::vector<int>* lengths
e.g.
https://google.github.io/styleguide/cppguide.html#Output_Parameters

again: it's not a hard rule, but I think it improved readability for generations to come :)

Reply:
Done!